gemini/sample-apps/fashion-trends-prediction/app/data_prep/driver.py

```diff
@@ -0,0 +1,331 @@
+                        if cat in saved["finaldata"][country_name]:
+                            saved["finaldata"][country_name][cat].extend(
+                                image_attr[cat].copy()
+                            )
+                        else:
+                            saved["finaldata"][country_name][cat] = image_attr[
+                                cat
+                            ].copy()
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def merge_keys():
+    """Merges keys in the final data file that have only a difference of spaces, capital letters, etc."""
+
+    # merging the keys which only have a difference of spaces, capital letters
+
+    with open(data_path, "r") as f:
+        saved = json.load(f)
+
+    print("merging keys")
+    finaldata_new = {}
+    finaldata = saved["finaldata"]
+    countries = [key for key in finaldata]
+
+    for country in countries:
+        finaldata_new[country] = {}
+
+        for key in finaldata[country]:
+            if isRoot(key):
+                finaldata_new[country][key] = finaldata[country][key].copy()
+
+        for key in finaldata[country]:
+            if not isRoot(key):
+                newkey = (key.strip()).lower()
+                if newkey in finaldata_new[country]:
+                    finaldata_new[country][newkey] += finaldata[country][key].copy()
+                else:
+                    finaldata_new[country][newkey] = finaldata[country][key].copy()
+
+    saved["finaldata"] = finaldata_new
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def combine_insta_and_vogue():
+    """Combines data from Instagram and Vogue into a single file."""
+
+    # Combining Instagram (India) and Vogue-India data
+
+    with open(data_path, "r") as f:
+        saved = json.load(f)
+
+    a = saved["finaldata"]["India"]
+    a_articles = saved["article_attributes"]
+    combined_attributes = {}
+
+    a2 = {}
+    for key in a:
+        temp = a[key]
+        newlist = [[item, "", ""] for item in temp]
+        a2[key] = newlist
+
+    for key in a2:
+        if key in a_articles:
+            combined_attributes[key] = a2[key] + a_articles[key]
+        else:
+            combined_attributes[key] = a2[key]
+
+    for key in a_articles:
+        if key not in a2:
+            combined_attributes[key] = a_articles[key]
+
+    combined_attributes["jewellery"] = a_articles["jewellery"]
+
+    saved["finaldata"]["India"] = combined_attributes
+
+    # Make data of all countries in the format of India's data
+    for country in saved["finaldata"]:
+        if country != "India":
+            for key in saved["finaldata"][country]:
+                temp = saved["finaldata"][country][key]
+                saved["finaldata"][country][key] = [[item, "", ""] for item in temp]
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def create_news_articles_data():
```

Comment:
Does this function need to exist? Could it be combined with `get_articles`?

Reply:
Yes removed this function.